File: notifications/src/main/java/com/cozy/notifications/config/KafkaConfig.java
package com.cozy.notifications.config;

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.context.annotation.Configuration;
import org.springframework.kafka.annotation.EnableKafka;

@Configuration
@ConditionalOnProperty(name = "app.notification-service", havingValue = "kafka", matchIfMissing = true)
@EnableKafka
public class KafkaConfig {
}

File: linkease/server/src/main/kotlin/com/linkease/server/kafka/KafkaConfig.kt
package com.linkease.server.kafka

import com.linkease.server.config.AppProperties
import org.apache.kafka.clients.admin.NewTopic
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.kafka.annotation.EnableKafka
import org.springframework.kafka.config.TopicBuilder

@EnableKafka
@Configuration
class KafkaConfig(private val appProperties: AppProperties) {

    @Bean
    fun notificationTopic(): NewTopic =
        TopicBuilder.name(appProperties.notificationTopic).partitions(1).replicas(1).build()
}
